src/main_win.rs

```diff
@@ -44,6 +44,24 @@ pub struct MainState {
     pub config: Rc<RefCell<Config>>,
 }
 
+impl MainState {
+    fn new(config: &Rc<RefCell<Config>>) -> MainState {
+        let theme_name = crate::pref_storage::get_theme_schema();
+        debug!("{}: {}", gettext("Theme name"), &theme_name);
+
+        MainState {
+            themes: Default::default(),
+            theme_name,
+            theme: Default::default(),
+            styles: Default::default(),
+            fonts: Default::default(),
+            avail_languages: Default::default(),
+            selected_language: Default::default(),
+            config: config.clone(),
```

Comment:
Why clone it here instead of taking ownership?

Reply:
It would require some restructuring, because config is used to connect actions later on in the function, but I'm sure it's possible.

Reply:
The easiest would be to just pass config.clone() into Mainstate::new(), I suppose

Reply:
Now I'm wondering though, is there a reason to reference count config, if we just pass its ownership to main_state? That might simplify some things a bit.

Reply:
We don't pass the ownership of the Config around, we pass the ownership of the RefCell around

Reply:
(We need Config both for EditView and PrefsWin and both mutable and non-mutable, so normal refs won't do)

Reply:
But can't we access it through main_state, which is reference counted?

Reply:
Right, that'd work!

Reply:
Good thinking :)